src/mmap-go-prototype/mmap.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"os"
+	"sync/atomic"
+	"syscall"
+	"unsafe"
+)
+
+type Mmap struct {
+	data []byte
+}
+
+func NewMmap(f string) *Mmap {
+	file, err := os.OpenFile(f, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	pageSize := syscall.Getpagesize()
+
+	// XXX: Passing `FALLOC_FL_ZERO_RANGE` as mode would zero the file, but
+	// I couldn't find this constant in `syscall`...
+	mode := uint32(0)
+
+	if err := syscall.Fallocate(int(file.Fd()), mode, 0, int64(pageSize)); err != nil {
```

Comment:
This line fails to compile on mac
```
./mmap.go:30:12: undefined: syscall.Fallocate
```

Reply:
Ok, lets try to find a workaround for this.